Library/Homebrew/cask/cask.rb

```diff
@@ -158,6 +159,12 @@ def caskfile_only?
       languages.any? || artifacts.any?(Artifact::AbstractFlightBlock)
     end
 
+    def uninstall_flight_blocks?
+      artifacts.any? do |artifact|
+        artifact.is_a?(Artifact::AbstractFlightBlock) && artifact.uninstall?
+      end
+    end
```

Comment:
Idea: Could we reuse this logic to be more explicit about matching uninstall flight blocks?

https://github.com/Homebrew/brew/blob/8c4c7319fc6ba3a69b1ba65659b03d418ebbfb2f/Library/Homebrew/cask/audit.rb#L164-L174

Reply:
```suggestion
    def uninstall_flight_blocks?
      artifacts.any? do |artifact|
        case artifact
        when Artifact::PreflightBlock
          artifact.directives.key?(:uninstall_preflight)
        when Artifact::PostflightBlock
          artifact.directives.key?(:uninstall_postflight)
        end
      end
    end
```

This might potentially work but I have not tested it locally.